dev_guide/migrating_applications/database_applications.adoc

```diff
@@ -0,0 +1,382 @@
+database.
++
+----
+$ mysql -u root
+$ source all.sql
+----
++
+Grant all privileges on *<dbname>* to `<your_v2_username>@localhost`, then flush
+privileges.
+
+. Remove the dump directory from the pod:
++
+----
+$ cd ../; rm -rf /var/lib/mysql/data/mysqldumpdir
+----
+
+[[migrating-database-applications-mysql-supported-environment-variables]]
+*Supported MySQL Environment Variables*
+
+[cols="2,2",options="header"]
+|===
+|v2 |v3
+
+|`*OPENSHIFT_MYSQL_DB_HOST*`   |`*[service_name]_SERVICE_HOST*`
+
+|`*OPENSHIFT_MYSQL_DB_PORT*`     |`*[service_name]_SERVICE_PORT*`
+
+|`*OPENSHIFT_MYSQL_DB_USERNAME*` |`*MYSQL_USER*`
+
+|`*OPENSHIFT_MYSQL_DB_PASSWORD*`  |`*MYSQL_PASSWORD*`
+
+|`*OPENSHIFT_MYSQL_DB_URL*`  |
+
+|`*OPENSHIFT_MYSQL_DB_LOG_DIR*`  |
+
+|`*OPENSHIFT_MYSQL_VERSION*`  |
+
+|`*OPENSHIFT_MYSQL_DIR*`  |
+
+|`*OPENSHIFT_MYSQL_DB_SOCKET*`  |
+
+|`*OPENSHIFT_MYSQL_IDENT*`  |
+
+|`*OPENSHIFT_MYSQL_AIO*`  |`*MYSQL_AIO*`
+
+|`*OPENSHIFT_MYSQL_MAX_ALLOWED_PACKET*`  |`*MYSQL_MAX_ALLOWED_PACKET*`
+
+|`*OPENSHIFT_MYSQL_TABLE_OPEN_CACHE*`  |`*MYSQL_TABLE_OPEN_CACHE*`
+
+|`*OPENSHIFT_MYSQL_SORT_BUFFER_SIZE*`  |`*MYSQL_SORT_BUFFER_SIZE*`
+
+|`*OPENSHIFT_MYSQL_LOWER_CASE_TABLE_NAMES*`  |`*MYSQL_LOWER_CASE_TABLE_NAMES*`
+
+|`*OPENSHIFT_MYSQL_MAX_CONNECTIONS*`  |`*MYSQL_MAX_CONNECTIONS*`
+
+|`*OPENSHIFT_MYSQL_FT_MIN_WORD_LEN*`  |`*MYSQL_FT_MIN_WORD_LEN*`
+
+|`*OPENSHIFT_MYSQL_FT_MAX_WORD_LEN*`  |`*MYSQL_FT_MAX_WORD_LEN*`
+
+|`*OPENSHIFT_MYSQL_DEFAULT_STORAGE_ENGINE*`  |
+
+|`*OPENSHIFT_MYSQL_TIMEZONE*`  |
+
+|  |`*MYSQL_DATABASE*`
+
+|  |`*MYSQL_ROOT_PASSWORD*`
+
+|  |`*MYSQL_MASTER_USER*`
+
+|  |`*MYSQL_MASTER_PASSWORD*`
+|===
+
+[[dev-guide-migrating-database-applications-postgresql]]
+== PostgreSQL
+
+. Back up the v2 PostgreSQL database from the gear:
++
+----
+$ rhc ssh -a <v2-application_name>
+$ mkdir ~/app-root/data/tmp
+$ pg_dump <database_name> | gzip > ~/app-root/data/tmp/<database_name>.gz
+----
+
+. Extract the backup file back to your local machine:
++
+----
+$ rhc scp -a <v2_application_name> download <local_dest> app-root/data/tmp/<db-name>.gz
+$ gzip -d <database-name>.gz
+----
++
+[NOTE]
+====
+Save the backup file to a separate folder for step 4.
+====
+
+. Create the PostgreSQL service using the v2 application database name, user name
+and password to create the new service:
++
+----
+$ oc new-app postgresql-persistent -p POSTGRESQL_DATABASE=dbname -p
+POSTGRESQL_PASSWORD=password -p POSTGRESQL_USER=username
+----
+
+. Check to see if the pod is ready to use:
++
+----
+$ oc get pods
+----
+
+ . When the pod is up and running, sync the backup directory to pod:
++
+----
+$ oc rsync /local/path/to/dir <postgresql_pod_name>:/var/lib/pgsql/data
+----
+
+. Remotely access the pod:
++
+----
+$ oc rsh <pod_name>
+----
+
+. Restore the database:
++
+----
+psql dbname < /var/lib/pgsql/data/<database_backup_file>
+----
+
+. Remove all backup files that are no longer needed:
++
+----
+$ rm < /var/lib/pgsql/data/<database-backup-file>
```

Comment:
Looks strange because `<` means redirection of the file descriptor. And `rm` doesn't need it.